File: android/app/src/main/java/be/sportgreenmoove/app/ui/ProfileScreen.kt
package be.sportgreenmoove.app.ui

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp
import be.sportgreenmoove.app.data.AppRole
import be.sportgreenmoove.app.data.ImpactSummary
import be.sportgreenmoove.app.data.RewardSummary
import be.sportgreenmoove.app.design.Sgm
import be.sportgreenmoove.app.design.SgmColor
import be.sportgreenmoove.app.design.SgmGridTexture
import be.sportgreenmoove.app.design.SgmRadius
import be.sportgreenmoove.app.design.SgmType
import java.util.Locale

private fun profileSettings(primaryClubLabel: String) = listOf(
    ProfileSetting(SgmIcon.Groups, "Mon club", primaryClubLabel, ProfileAction.Groups),
    ProfileSetting(SgmIcon.Award, "Paiements", "Stripe", ProfileAction.Payments),
    ProfileSetting(SgmIcon.Location, "Ma ville", "Wavre, Belgique", ProfileAction.Options),
    ProfileSetting(SgmIcon.Bell, "Notifications", "Activées", ProfileAction.Options),
    ProfileSetting(SgmIcon.Settings, "Paramètres", "", ProfileAction.Options),
)

@Composable
fun ProfileScreen(
    role: AppRole,
    displayName: String?,
    email: String?,
    primaryClubLabel: String,
    impactSummary: ImpactSummary,
    rewardSummary: RewardSummary,
    onRoleChange: (AppRole) -> Unit,
    onGroups: () -> Unit,
    onImpact: () -> Unit,
    onRewards: () -> Unit,
    onPayments: () -> Unit,
    onOptions: () -> Unit,
    onLogout: () -> Unit,
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Sgm.colors.bgApp)
            .verticalScroll(rememberScrollState())
            .padding(bottom = 20.dp),
    ) {
        V2TopBar("MON PROFIL")
        ProfileIdentity(displayName = displayName, email = email, primaryClubLabel = primaryClubLabel)
        ProfileRoleSelector(role = role, onRoleChange = onRoleChange)
        ProfileImpactCard(summary = impactSummary, onClick = onImpact)
        ProfileRewardsCard(summary = rewardSummary, onClick = onRewards)
        V2SectionLabel("PARAMÈTRES")
        ProfileSettingsCard(
            primaryClubLabel = primaryClubLabel,
            onGroups = onGroups,
            onPayments = onPayments,
            onOptions = onOptions,
            modifier = Modifier.padding(horizontal = 20.dp),
        )
        Box(Modifier.padding(start = 20.dp, top = 20.dp, end = 20.dp)) {
            V2Button("SE DÉCONNECTER", onClick = onLogout, variant = V2ButtonVariant.Ghost, full = true)
        }
        Spacer(Modifier.height(76.dp))
    }
}

@Composable
private fun ProfileIdentity(displayName: String?, email: String?, primaryClubLabel: String) {
    val name = displayName?.takeIf(String::isNotBlank) ?: email?.substringBefore("@") ?: "Green-Mover"
    val initials = name.split(Regex("\\s+"))
        .filter(String::isNotBlank)
        .take(2)
        .joinToString("") { it.first().uppercaseChar().toString() }
        .ifBlank { "GM" }
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(start = 20.dp, top = 16.dp, end = 20.dp, bottom = 20.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        Box(
            modifier = Modifier
                .size(80.dp)
                .shadow(10.dp, CircleShape, clip = false)
                .clip(CircleShape)
                .background(Brush.linearGradient(listOf(SgmColor.Green, SgmColor.GreenDark))),
            contentAlignment = Alignment.Center,
        ) {
            Text(initials, style = SgmType.Display2XL.copy(color = SgmColor.TextOnGreen, fontSize = 30.sp, letterSpacing = 0.04.em))
        }
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text(
                name.uppercase(Locale.FRANCE),
                style = SgmType.DisplayXL.copy(color = Sgm.colors.textPrimary, fontSize = 22.sp, letterSpacing = 0.06.em),
            )
            Text(
                "$primaryClubLabel${email?.let { " · $it" }.orEmpty()}",
                style = SgmType.BodySM.copy(color = Sgm.colors.textMuted, fontSize = 13.sp, fontWeight = FontWeight.Medium),
            )
        }
        Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            ProfilePill("GREEN-MOOVER", selected = false)
            ProfilePill("U8 NATIONAUX", selected = true)
        }
    }
}

@Composable
private fun ProfileImpactCard(summary: ImpactSummary, onClick: () -> Unit) {
    Box(
        modifier = Modifier
            .padding(horizontal = 20.dp)
            .fillMaxWidth()
            .clip(RoundedCornerShape(20.dp))
            .background(SgmColor.HeroGradient)
            .clickable(onClick = onClick)
            .padding(horizontal = 20.dp, vertical = 18.dp),
    ) {
        SgmGridTexture()
        Column {
            Text(
                "MON IMPACT CO₂",
                style = SgmType.Label.copy(color = SgmColor.TextOnGreen.copy(alpha = 0.50f), fontSize = 11.sp, letterSpacing = 0.14.em),
            )
            Row(
                modifier = Modifier.padding(top = 6.dp),
                verticalAlignment = Alignment.Bottom,
                horizontalArrangement = Arrangement.spacedBy(24.dp),
            ) {
                Column {
                    Text(kgValue(summary.totalCo2Kg), style = SgmType.Display4XL.copy(color = SgmColor.Green, fontSize = 56.sp))
                    Text(
                        "kg CO₂ économisés",
                        style = SgmType.BodySM.copy(color = SgmColor.TextOnGreen.copy(alpha = 0.60f), fontWeight = FontWeight.SemiBold),
                    )
                }
                Column(
                    modifier = Modifier.padding(bottom = 4.dp),
                    verticalArrangement = Arrangement.spacedBy(4.dp),
                ) {
                    Text("${summary.sharedDistanceKm} km partagés", style = ProfileImpactMetaStyle())
                    Text("${summary.rideCount} trajets clôturés", style = ProfileImpactMetaStyle())
                }
            }
        }
    }
}

@Composable
private fun ProfileRewardsCard(summary: RewardSummary, onClick: () -> Unit) {
    Column(
        modifier = Modifier
            .padding(start = 20.dp, top = 16.dp, end = 20.dp, bottom = 16.dp)
            .fillMaxWidth()
            .clip(RoundedCornerShape(20.dp))
            .background(Sgm.colors.bgCard)
            .border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(20.dp))
            .clickable(onClick = onClick)
            .padding(horizontal = 16.dp, vertical = 14.dp),
        verticalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            SgmLineIcon(icon = SgmIcon.Award, tint = SgmColor.Orange, modifier = Modifier.size(22.dp))
            Text(
                "RÉCOMPENSES",
                style = SgmType.DisplayLG.copy(color = Sgm.colors.textPrimary, fontSize = 18.sp, letterSpacing = 0.06.em),
                modifier = Modifier
                    .padding(start = 8.dp)
                    .weight(1f),
            )
            Text(moneyLabel(summary.balanceCents), style = SgmType.DisplayXL.copy(color = SgmColor.Orange, fontSize = 24.sp))
        }
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(6.dp)
                .clip(RoundedCornerShape(999.dp))
                .background(Sgm.colors.border),
        ) {
            Box(
                modifier = Modifier
                    .fillMaxWidth(summary.progress.coerceIn(0f, 1f))
                    .height(6.dp)
                    .clip(RoundedCornerShape(999.dp))
                    .background(Brush.horizontalGradient(listOf(SgmColor.Green, SgmColor.Orange))),
            )
        }
        Row {
            Text("Prochain palier à ${moneyLabel(summary.nextTierCents)}", style = SgmType.BodyXS.copy(color = Sgm.colors.textMuted, fontSize = 11.sp, fontWeight = FontWeight.Medium), modifier = Modifier.weight(1f))
            Text(percentLabel(summary.progress), style = SgmType.BodyXS.copy(color = Sgm.colors.textMuted, fontSize = 11.sp, fontWeight = FontWeight.Medium))
        }
    }
}

@Composable
private fun ProfileSettingsCard(
    primaryClubLabel: String,
    onGroups: () -> Unit,
    onPayments: () -> Unit,
    onOptions: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val rows = profileSettings(primaryClubLabel)
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(SgmRadius.LG))
            .background(Sgm.colors.bgSurface)
            .border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(SgmRadius.LG)),
    ) {
        rows.forEachIndexed { index, setting ->
            ProfileSettingRow(
                setting = setting,
                showDivider = index < rows.lastIndex,
                onClick = when (setting.action) {
                    ProfileAction.Groups -> onGroups
                    ProfileAction.Payments -> onPayments
                    ProfileAction.Options -> onOptions
                },
            )
        }
    }
}

@Composable
private fun ProfileSettingRow(setting: ProfileSetting, showDivider: Boolean, onClick: () -> Unit) {
    Column(modifier = Modifier.clickable(onClick = onClick)) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 16.dp, vertical = 13.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            SgmLineIcon(icon = setting.icon, tint = Sgm.colors.textMuted, modifier = Modifier.size(18.dp))
            Text(setting.label, style = SgmType.BodyBase.copy(color = Sgm.colors.textPrimary, fontSize = 14.sp, fontWeight = FontWeight.SemiBold), modifier = Modifier.weight(1f))
            if (setting.value.isNotEmpty()) {
                Text(setting.value, style = SgmType.BodySM.copy(color = Sgm.colors.textMuted, fontSize = 13.sp, fontWeight = FontWeight.Medium))
            }
            SgmLineIcon(icon = SgmIcon.ChevronRight, tint = Sgm.colors.textMuted, modifier = Modifier.size(16.dp))
        }
        if (showDivider) Box(Modifier.padding(start = 46.dp).fillMaxWidth().height(1.dp).background(Sgm.colors.border))
    }
}

@Composable
private fun ProfilePill(text: String, selected: Boolean) {
    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(999.dp))
            .background(if (selected) SgmColor.Green else Sgm.colors.bgCard)
            .then(if (selected) Modifier else Modifier.border(BorderStroke(1.dp, Sgm.colors.border), RoundedCornerShape(999.dp)))
            .padding(horizontal = 12.dp, vertical = 3.dp),
    ) {
        Text(text, style = SgmType.Label.copy(color = if (selected) SgmColor.TextOnGreen else SgmColor.Green, fontSize = 11.sp, letterSpacing = 0.04.em))
    }
}

private fun ProfileImpactMetaStyle() = SgmType.BodyXS.copy(color = SgmColor.TextOnGreen.copy(alpha = 0.70f), fontSize = 12.sp)
private fun kgValue(value: Double): String = String.format(Locale.FRANCE, "%.1f", value)
private fun moneyLabel(cents: Int): String {
    val sign = if (cents < 0) "-" else ""
    val abs = kotlin.math.abs(cents)
    return String.format(Locale.FRANCE, "%s%d,%02d€", sign, abs / 100, abs % 100)
}
private fun percentLabel(progress: Float): String = "${(progress.coerceIn(0f, 1f) * 100).toInt()}%"
private data class ProfileSetting(val icon: SgmIcon, val label: String, val value: String, val action: ProfileAction)
private enum class ProfileAction { Groups, Payments, Options }

File: ios/Sources/SportsGreenMooveApp/ProfileScreen.swift
import SwiftUI

struct ProfileScreen: View {
    @Environment(AppState.self) private var appState

    var body: some View {
        SGMScreen(spacing: 0) {
            SGMTopBar(title: "MON PROFIL")
            ProfileIdentity()
            ProfileRoleSelector()
            ProfileImpactCard {
                appState.openOverlay(.impact)
            }
            ProfileRewardsCard {
                appState.openOverlay(.rewards)
            }
            SGMSectionLabel("PARAMÈTRES")
            ProfileSettingsCard()
                .padding(.horizontal, SGMSpace.padScreen)
            SGMButton(title: "SE DÉCONNECTER", variant: .ghost) {}
                .padding(.horizontal, SGMSpace.padScreen)
                .padding(.top, 4)
        }
    }
}

private struct ProfileRoleSelector: View {
    @Environment(AppState.self) private var appState

    var body: some View {
        VStack(spacing: 6) {
            HStack(spacing: 6) {
                RoleChip("Parent", role: .parent)
                RoleChip("Conducteur", role: .driver)
            }
            HStack(spacing: 6) {
                RoleChip("Enfant", role: .child)
                RoleChip("Club manager", role: .clubManager)
            }
        }
        .padding(.horizontal, SGMSpace.padScreen)
        .padding(.bottom, 14)
    }
}

private struct RoleChip: View {
    @Environment(AppState.self) private var appState
    let title: String
    let role: AppRole

    init(_ title: String, role: AppRole) {
        self.title = title
        self.role = role
    }

    var body: some View {
        SGMChip(text: title, selected: appState.selectedRole == role) {
            appState.selectedRole = role
            Task { await appState.refreshAppData() }
        }
    }
}

private struct ProfileIdentity: View {
    @Environment(AppState.self) private var appState

    var body: some View {
        VStack(spacing: 10) {
            Text("OB")
                .font(.sgmDisplay(30))
                .tracking(.sgmWide(for: 30))
                .foregroundStyle(SGM.textOnGreen)
                .frame(width: 80, height: 80)
                .background(
                    LinearGradient(colors: [SGM.green, SGM.greenDark], startPoint: .topLeading, endPoint: .bottomTrailing),
                    in: Circle()
                )
                .shadow(color: SGM.green.opacity(0.24), radius: 10, y: 4)

            VStack(spacing: 2) {
                Text("BAKOMBA-NZOLUVONDA")
                    .font(.sgmDisplay(22))
                    .tracking(.sgmWide(for: 22))
                    .foregroundStyle(SGM.textPrimary)
                Text("Olivier · \(primaryClubLabel)")
                    .font(.sgmBody(13, weight: .medium))
                    .foregroundStyle(SGM.textMuted)
            }

            HStack(spacing: 6) {
                ProfilePill("GREEN-MOOVER", selected: false)
                ProfilePill("U8 NATIONAUX", selected: true)
            }
        }
        .frame(maxWidth: .infinity)
        .padding(.horizontal, SGMSpace.padScreen)
        .padding(.top, 16)
        .padding(.bottom, 10)
    }

    private var primaryClubLabel: String {
        appState.clubSummaries.first(where: \.isMember)?.name ?? "Aucun club lié"
    }
}

private struct ProfileImpactCard: View {
    @Environment(AppState.self) private var appState
    let action: () -> Void

    var body: some View {
        let summary = appState.impactSummary
        Button(action: action) {
            ZStack(alignment: .leading) {
                SGM.heroGradient
                SGMGridTexture()
                VStack(alignment: .leading, spacing: 6) {
                    Text("MON IMPACT CO₂")
                        .font(.sgmBody(11, weight: .bold))
                        .tracking(.sgmWider(for: 11))
                        .foregroundStyle(SGM.textOnGreen.opacity(0.5))
                    HStack(alignment: .bottom, spacing: 24) {
                        VStack(alignment: .leading, spacing: 0) {
                            Text(profileKgValue(summary.totalCo2Kg))
                                .font(.sgmDisplay(56))
                                .foregroundStyle(SGM.green)
                            Text("kg CO₂ économisés")
                                .font(.sgmBody(13, weight: .semibold))
                                .foregroundStyle(SGM.textOnGreen.opacity(0.6))
                        }
                        VStack(alignment: .leading, spacing: 4) {
                            Text("\(summary.sharedDistanceKm) km partagés")
                            Text("\(summary.rideCount) trajets clôturés")
                        }
                        .font(.sgmBody(12))
                        .foregroundStyle(SGM.textOnGreen.opacity(0.72))
                    }
                }
                .padding(.horizontal, 20)
                .padding(.vertical, 18)
            }
            .clipShape(RoundedRectangle(cornerRadius: 20, style: .continuous))
        }
        .buttonStyle(.plain)
        .padding(.horizontal, SGMSpace.padScreen)
    }
}

private struct ProfileRewardsCard: View {
    @Environment(AppState.self) private var appState
    let action: () -> Void

    var body: some View {
        let summary = appState.rewardSummary
        Button(action: action) {
            VStack(spacing: 10) {
                HStack {
                    SGMIconView(icon: .award, size: 22, color: SGM.orange)
                    Text("RÉCOMPENSES")
                        .font(.sgmDisplay(18))
                        .tracking(.sgmWide(for: 18))
                        .foregroundStyle(SGM.textPrimary)
                    Spacer()
                    Text(profileMoneyLabel(summary.balanceCents))
                        .font(.sgmDisplay(24))
                        .foregroundStyle(SGM.orange)
                }
                SGMProgressBar(progress: CGFloat(summary.progress))
                HStack {
                    Text("Prochain palier à \(profileMoneyLabel(summary.nextTierCents))")
                    Spacer()
                    Text(profilePercentLabel(summary.progress))
                }
                .font(.sgmBody(11, weight: .medium))
                .foregroundStyle(SGM.textMuted)
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 14)
            .background(SGM.bgCard, in: RoundedRectangle(cornerRadius: 20, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 20, style: .continuous).stroke(SGM.border, lineWidth: 1))
        }
        .buttonStyle(.plain)
        .padding(.horizontal, SGMSpace.padScreen)
        .padding(.top, 6)
    }
}

private struct ProfileSettingsCard: View {
    @Environment(AppState.self) private var appState

    var body: some View {
        VStack(spacing: 0) {
            ForEach(Array(rows.enumerated()), id: \.element.id) { index, row in
                Button {
                    appState.openOverlay(row.destination)
                } label: {
                    HStack(spacing: 12) {
                        SGMIconView(icon: row.icon, size: 18, color: SGM.textMuted)
                        Text(row.label)
                            .font(.sgmBody(14, weight: .semibold))
                            .foregroundStyle(SGM.textPrimary)
                        Spacer()
                        if !row.value.isEmpty {
                            Text(row.value)
                                .font(.sgmBody(13, weight: .medium))
                                .foregroundStyle(SGM.textMuted)
                        }
                        SGMIconView(icon: .chevronRight, size: 13, color: SGM.textMuted)
                    }
                    .padding(.horizontal, 16)
                    .padding(.vertical, 13)
                }
                .buttonStyle(.plain)
                if index < rows.count - 1 {
                    Rectangle()
                        .fill(SGM.border)
                        .frame(height: 1)
                        .padding(.leading, 46)
                }
            }
        }
        .background(SGM.bgSurface, in: RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: SGMRadius.lg, style: .continuous).stroke(SGM.border, lineWidth: 1))
    }

    private var rows: [ProfileSetting] {
        [
            ProfileSetting(.groups, "Mon club", appState.clubSummaries.first(where: \.isMember)?.name ?? "Aucun club lié", AppOverlay.groups),
            ProfileSetting(.award, "Paiements", "Stripe", AppOverlay.payments),
            ProfileSetting(.location, "Ma ville", "Wavre, Belgique", AppOverlay.options),
            ProfileSetting(.bell, "Notifications", "Activées", AppOverlay.options),
            ProfileSetting(.settings, "Paramètres", "", AppOverlay.options)
        ]
    }
}

private struct ProfilePill: View {
    let text: String
    let selected: Bool

    init(_ text: String, selected: Bool) {
        self.text = text
        self.selected = selected
    }

    var body: some View {
        Text(text)
            .font(.sgmBody(11, weight: .bold))
            .tracking(.sgmWide(for: 11))
            .foregroundStyle(selected ? SGM.textOnGreen : SGM.green)
            .padding(.horizontal, 12)
            .padding(.vertical, 3)
            .background(selected ? SGM.green : SGM.bgCard, in: Capsule())
            .overlay(Capsule().stroke(selected ? .clear : SGM.border, lineWidth: 1))
    }
}

private struct ProfileSetting: Identifiable {
    let id = UUID()
    let icon: SGMIcon
    let label: String
    let value: String
    let destination: AppOverlay

    init(_ icon: SGMIcon, _ label: String, _ value: String, _ destination: AppOverlay) {
        self.icon = icon
        self.label = label
        self.value = value
        self.destination = destination
    }
}

private func profileKgValue(_ value: Double) -> String {
    let formatter = NumberFormatter()
    formatter.locale = Locale(identifier: "fr_FR")
    formatter.minimumFractionDigits = 1
    formatter.maximumFractionDigits = 1
    return formatter.string(from: NSNumber(value: value)) ?? String(format: "%.1f", value)
}

private func profileMoneyLabel(_ cents: Int) -> String {
    let sign = cents < 0 ? "-" : ""
    let absolute = abs(cents)
    return "\(sign)\(absolute / 100),\(String(format: "%02d", absolute % 100))€"
}

private func profilePercentLabel(_ progress: Double) -> String {
    "\(Int(min(1, max(0, progress)) * 100))%"
}
